float32 weight
float32 mu_x
float32 mu_y
float32 sigma_x
float32 sigma_y